src/fetchers/tarball-fetcher.js

```diff
@@ -175,13 +176,16 @@ export default class TarballFetcher extends BaseFetcher {
   }
 
   async _fetch(): Promise<FetchedOverride> {
+    const isFilePath = this.reference.startsWith('file:');
```

Comment:
copied from https://github.com/yarnpkg/yarn/pull/3709 for bundledDependencies case